src/screens/Refunds/RefundUtils.res

```diff
@@ -114,23 +139,102 @@ let initialFixedFilter = () => [
   ),
 ]
 
-let initialFilters = json => {
+let getConditionalFilter = (key, dict, filterValues) => {
+  open LogicUtils
+
+  let filtersArr = switch key->getFilterTypeFromString {
+  | #connector_label => {
+      let arr = filterValues->getArrayFromDict("connector", [])->getStrArrayFromJsonArray
+      let new_arr = []
+      arr->Array.forEach(item => {
+        let connectorLabelArr = dict->getDictfromDict("connector")->getArrayFromDict(item, [])
+        connectorLabelArr->Array.forEach(item => {
+          let connector_label = item->getDictFromJsonObject->getString("connector_label", "")
+          new_arr->Array.push(connector_label)
+        })
+      })
+      new_arr
+    }
+  | _ => []
+  }
+
+  filtersArr
+}
+
+let getMerchantIdforConnector: (
+  Js.Dict.t<'a>,
+  Js.Dict.t<'a>,
+) => array<FilterSelectBox.dropdownOption> = (dict, filterValues) => {
   open LogicUtils
+  let arr = filterValues->getArrayFromDict("connector", [])->getStrArrayFromJsonArray
+  let new_arr: array<FilterSelectBox.dropdownOption> = []
+  arr->Array.forEach(item => {
```

Comment:
use map

Reply:
i have used foreach because i am using map for two times then return type becomes array<array<t>> but i am expecting an array<t>, thats why i created an array and pushed items to that

Reply:
```
  let newArr = arr->Array.flatMap(it => {
    let connectorLabelArr = dict->getDictfromDict("connector")->getArrayFromDict(it, [])
    connectorLabelArr->Array.map(item => {
      let label = item->getDictFromJsonObject->getString("connector_label", "")
      let value = item->getDictFromJsonObject->getString("merchant_connector_id", "")
      let option: FilterSelectBox.dropdownOption = {
        label,
        value,
      }
      option
    })
  })
```
can you check this

Reply:
yes, working fine. thanks